sdks/python/apache_beam/io/fileio.py

```diff
@@ -562,16 +569,21 @@ def process(self, element):
                        r.pane,
                        destination)
 
-    logging.info('Cautiously removing temporary files for destination %s',
-                 destination)
-    self._remove_temporary_files([r.file_name for r in file_results])
+    logging.info('Cautiously removing temporary files for'
+                 ' destination %s and window %s', destination, w)
+    writer_key = (destination, w)
+    self._remove_temporary_files(writer_key)
 
-  @staticmethod
-  def _remove_temporary_files(files):
+  def _remove_temporary_files(self, writer_key):
     try:
-      filesystems.FileSystems.delete(files)
+      prefix = filesystems.FileSystems.join(
+          self.temporary_directory.get(), str(abs(hash(writer_key))))
+      match_result = filesystems.FileSystems.match(['%s*' % prefix])
+      orphaned_files = [m.path for m in match_result[0].metadata_list]
+
+      logging.debug('Deleting orphaned files: %s', orphaned_files)
+      filesystems.FileSystems.delete(orphaned_files)
```

Comment:
Shouldn't we be ignoring missing source files to handle workitem retries ?

Reply:
I agree that we should ignore errors here. Note that `orphaned_files` are taken from a recent `filesystems.FileSystems.match()` call, so it doesn't make a difference if this is a retry.

Reply:
Yes. All we're doing is logging the failures (in debug level) to be aware of them.